src/test/ui/span/issue-27522.stderr

```diff
@@ -1,11 +1,20 @@
-error[E0308]: mismatched method receiver
+error[E0307]: invalid `self` type: &SomeType
   --> $DIR/issue-27522.rs:16:22
    |
 16 |     fn handler(self: &SomeType);
-   |                      ^^^^^^^^^ expected Self, found struct `SomeType`
+   |                      ^^^^^^^^^
    |
-   = note: expected type `&Self`
-              found type `&SomeType`
+   = note: type must be `Self` or a type that dereferences to it`
+   = help: consider changing to `self`, `&self`, `&mut self`, or `self: Box<Self>`
 
-error: aborting due to previous error
+error: arbitrary `self` types are unstable (see issue #44874)
```

Comment:
Could you not have this error if you already emitted an "invalid self type" error?

Reply:
Currently it emits both errors. I can change it though, wasn't sure whether to emit both or just the "invalid `self` type" error

Reply:
You should change it to just emit the "invalid self type" error.